mrp_bom_component_cost/tests/test_mrp_bom_component_cost.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2025 Alberto Martínez <alberto.martinez@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import TransactionCase
+
+
+class TestMrpBomComponentCost(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.product = cls.env["product.product"].create(
+            {
+                "name": "Test Product",
+                "detailed_type": "product",
+            }
+        )
+        cls.component1 = cls.env["product.product"].create(
+            {"name": "Test Component", "detailed_type": "product", "standard_price": 1}
+        )
+        cls.component2 = cls.env["product.product"].create(
+            {
+                "name": "Test Component 2",
+                "detailed_type": "product",
+                "standard_price": 12,
+                "uom_id": cls.env.ref("uom.product_uom_dozen").id,
+            }
+        )
+        cls.bom = cls.env["mrp.bom"].create(
+            {
+                "product_id": cls.product.id,
+                "product_tmpl_id": cls.product.product_tmpl_id.id,
+                "bom_line_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": cls.component1.id,
+                            "product_qty": 2,
+                            "product_uom_id": cls.env.ref("uom.product_uom_dozen").id,
+                        },
+                    ),
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": cls.component2.id,
+                            "product_qty": 1,
+                            "product_uom_id": cls.env.ref("uom.product_uom_unit").id,
+                        },
+                    ),
+                ],
+            }
+        )
+
+    def test_mrp_bom_component_cost(self):
```

Comment:
Please, add a test with multicurrency.

Reply:
Are you sure? The currency is the product's one, that is the company's one.

Having multicurrency seems very rare to me, and currently I don't know how to functionaly configure it

@manuelregidor

Reply:
@Tisho99 

1. You've got "Product 1" with no company associated, but the main company uses EUR as currency.
2. You've got "Product 2" related to "Company 2", which uses USD as its currency.
3. You create a MRP Bom in Company 2 with Product 1 and Product 2. As the Company 2 uses USD, you should make sure that the cost for Product 1 is converted to USD too.